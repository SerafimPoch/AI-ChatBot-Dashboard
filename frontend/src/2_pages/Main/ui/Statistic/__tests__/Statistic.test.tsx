import React from "react";
import { render } from "@testing-library/react";
import {
  StatisticContainer,
  StatisticBoyIconWrapper,
  StatisticTitleWrapper,
  StatisticTitle,
  ViewEarningBtn,
} from "../Statistic.styles";

describe("Statistic page components", () => {
  it("StatisticContainer renders correctly", () => {
    const { asFragment } = render(<StatisticContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("StatisticBoyIconWrapper renders correctly", () => {
    const { asFragment } = render(<StatisticBoyIconWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("StatisticTitleWrapper renders correctly", () => {
    const { asFragment } = render(<StatisticTitleWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("StatisticTitle renders correctly", () => {
    const { asFragment } = render(<StatisticTitle />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("ViewEarningBtn renders correctly", () => {
    const { asFragment } = render(<ViewEarningBtn />);
    expect(asFragment()).toMatchSnapshot();
  });
});
