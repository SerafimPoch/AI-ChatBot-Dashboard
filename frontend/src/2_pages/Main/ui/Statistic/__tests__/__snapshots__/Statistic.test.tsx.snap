// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistic page components StatisticBoyIconWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: absolute;
  right: -80px;
  top: -150px;
}

@media (max-width: 1000px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Statistic page components StatisticContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-radius: 32px;
  padding: 12px;
  position: relative;
  height: 100%;
  background: linear-gradient(
      245deg,
      #d6dcd1 57.34%,
      #ffddb4 81.64%,
      #c7b4ff 105.59%
    );
  grid-row: 1/span 2;
}

@media (max-width: 1550px) {
  .emotion-0 {
    grid-row: 1;
    grid-column: 1/3;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Statistic page components StatisticTitle renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-display);
  font-size: 64px;
  font-weight: 700;
  line-height: 64px;
  color: var(--neutral-05);
  margin-bottom: 24px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Statistic page components StatisticTitleWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 36px;
  margin-bottom: 46px;
}

@media (max-width: 1000px) {
  .emotion-0 {
    padding: 10px;
    margin-bottom: 0px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Statistic page components ViewEarningBtn renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 48px;
  padding: 12px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
  border-radius: 12px;
  background-color: undefined;
  color: undefined;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  font-family: var(--font-sf-pro-rounded);
}

.emotion-0:hover {
  opacity: 0.5;
}

<button
    class="emotion-0"
  />
</DocumentFragment>
`;
